### ISSUE TITLE
Compile problem with JSON Doc

### ISSUE THREAD

**Issue by wogoos** [bblanchon/ArduinoJson#1469]
Dear Blanchon,
I use your latest (14-11-20) JSON library v6 for a while, and it works nice for me on flat JSON Key-Value files.  I need to generate a more complicated JSON structure with arrays of JSON object which I manually made and checked on JSON syntax errors.  
I have used this JSON as input for https://arduinojson.org/v6/assistant/  (ESP8266, Serialze, Stream) and copied the code it generates in my program.  I have compiler errors on this piece of code. I use Visual studio Code as IDE. and the JSON library is located in the .lib directory.
I hope you can tell me what the problem is and what I need to do to correct it. 
Thanks Oscar Goos

The error is :
lib\ArduinoJson\src/ArduinoJson/Serialization/Writer.hpp:18:26: error: 'class ArduinoJson6172_91::ArrayRef' has no member named 'write'.
It's the line 5 and 8 return _dest->write(....);  of the file Writter.hpp  that generates this error.  The compiler complains about wite(...) is unknow to the compiler as far as I can see.
`
1   class Writer {
2   public:
3   explicit Writer(TDestination& dest) : _dest(&dest) {}

4    size_t write(uint8_t c) {
5    return _dest->write(c);
6    }

7    size_t write(const uint8_t* s, size_t n) {
8    return _dest->write(s, n);
9    }

10  private:
11  TDestination* _dest;
12  };`
```
My JSON is a configuration file for wifi[4] and serial ports[2].
{	"wifi": [{
		"DSid": "Wifi_cfg",
		"cmd": "ACK",
		"WIND": "2",
		"SSID": "INFINITUM8D73",
		"WPWS": "1894256060",
		"PHYM": "802.11N",
		"CHAN": "6",
		"TPWR": "19.5",
		"PROT": "WPA2",
		"KEY": "3",
		"HID": "0",
		"DHCP": "0",
		"NWIP": "192.168.1.80",
		"GWIP": "192.168.1.254",
		"MASK": "255.255.255.0",
		"DNS1": "8.26.56.26",
		"DNS2": "8.20.247.20"
	},{
		"DSid": "Wifi_cfg",
		"cmd": "ACK",
		"WIND": "2",
		"SSID": "INFINITUM8D73",
		"WPWS": "1894256060",
		"PHYM": "802.11N",
		"CHAN": "6",
		"TPWR": "19.5",
		"PROT": "WPA2",
		"KEY": "3",
		"HID": "0",
		"DHCP": "0",
		"NWIP": "192.168.1.80",
		"GWIP": "192.168.1.254",
		"MASK": "255.255.255.0",
		"DNS1": "8.26.56.26",
		"DNS2": "8.20.247.20"
	},{
		"DSid": "Wifi_cfg",
		"cmd": "ACK",
		"WIND": "2",
		"SSID": "INFINITUM8D73",
		"WPWS": "1894256060",
		"PHYM": "802.11N",
		"CHAN": "6",
		"TPWR": "19.5",
		"PROT": "WPA2",
		"KEY": "3",
		"HID": "0",
		"DHCP": "0",
		"NWIP": "192.168.1.80",
		"GWIP": "192.168.1.254",
		"MASK": "255.255.255.0",
		"DNS1": "8.26.56.26",
		"DNS2": "8.20.247.20"
	},{
		"DSid": "Wifi_cfg",
		"cmd": "ACK",
		"WIND": "2",
		"SSID": "INFINITUM8D73",
		"WPWS": "1894256060",
		"PHYM": "802.11N",
		"CHAN": "6",
		"TPWR": "19.5",
		"PROT": "WPA2",
		"KEY": "3",
		"HID": "0",
		"DHCP": "0",
		"NWIP": "192.168.1.80",
		"GWIP": "192.168.1.254",
		"MASK": "255.255.255.0",
		"DNS1": "8.26.56.26",
		"DNS2": "8.20.247.20"
	}],
	"Serial": [{
		"BD": "115200",
		"FRM": "8n1",
		"PRT": "1"
	},{
		"BD": "115200",
		"FRM": "8n1",
		"PRT": "1"
	}]
}

**Comment by wogoos** [comment]
Further investigation I  found a part of the problem. 
The compiler seems to crash on the bottom line serializeJson(doc, output); which is in my esp8266 case serializeJson(doc, Serial);
Without that line it compiles normal.

Further investigation I  found an other part of the problem. 
I have a function config() were I write the system variables to the SPIFFS as JSON file.  Initially declared the StaticJsonDocument<2400> doc; within that function. I figured out that with the size <1500> it worked and generated a serializeJson(doc, Serial);  but with size<2400> it crashed on this serializeJson(doc, Serial);. 
I decide to declare the StaticJsonDocument<2400> doc; as a global variable since I suspected that the stack was causing the problem beside 2400byte is quite big. I tested it as a global variable and that worked. Furthermore, I tested it with size<5000>and that compiled and executed normal too.

Question if some knows why I can't declare this StaticJsonDocument<2400> as a local variable within the config() function that would be nice to know.
